packages/amplify_datastore_plugin_interface/lib/src/types/models/model_field.dart

```diff
@@ -69,18 +69,16 @@ class ModelField {
   Map<String, dynamic> toMap() {
     Map<String, dynamic> map = {
       'name': name,
-      'type': type.toMap(),
+      'type': type!.toMap(),

```

Comment:
Shouldn't type be declared `required` since it's required 100% of the time?

Reply:
thanks updated it.  Change name, type, isRequired, and isArray to be non nullable.